pkg/mcpserver/server.go

```diff
@@ -2,53 +2,58 @@ package mcpserver
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"log"
 	"net/http"
-	"os"
 	"slices"
 	"strings"
 	"sync"
 
-	"github.com/mark3labs/mcp-go/mcp"
-	mcpserver "github.com/mark3labs/mcp-go/server"
+	"github.com/modelcontextprotocol/go-sdk/mcp"
 
+	"github.com/genmcp/gen-mcp/pkg/invocation"
+	_ "github.com/genmcp/gen-mcp/pkg/invocation/cli"
+	_ "github.com/genmcp/gen-mcp/pkg/invocation/http"
+	"github.com/genmcp/gen-mcp/pkg/invocation/utils"
 	"github.com/genmcp/gen-mcp/pkg/mcpfile"
 	"github.com/genmcp/gen-mcp/pkg/oauth"
 )
 
-func MakeServer(mcpServer *mcpfile.MCPServer) *mcpserver.MCPServer {
-	s := mcpserver.NewMCPServer(
-		mcpServer.Name,
-		mcpServer.Version,
-		mcpserver.WithToolCapabilities(true),
-		mcpserver.WithToolFilter(filterAuthorizedTools(mcpServer)),
-	)
-
-	for _, t := range mcpServer.Tools {
-		s.AddTool(
-			mcp.NewTool(
-				t.Name,
-				t.GetMCPToolOpts()...,
-			),
-			createAuthorizedToolHandler(t),
-		)
+func MakeServer(mcpServer *mcpfile.MCPServer) (*mcp.Server, error) {
+	// Validate the server configuration before creating the server
+	if err := mcpServer.Validate(invocation.InvocationValidator); err != nil {
+		return nil, fmt.Errorf("invalid server configuration: %w", err)
 	}
 
-	return s
+	return makeServerWithTools(mcpServer, mcpServer.Tools)
 }
 
 func RunServer(ctx context.Context, mcpServerConfig *mcpfile.MCPServer) error {
-	s := MakeServer(mcpServerConfig)
+	// Validate the server configuration before running
+	if err := mcpServerConfig.Validate(invocation.InvocationValidator); err != nil {
+		return fmt.Errorf("invalid server configuration: %w", err)
+	}
 
 	switch strings.ToLower(mcpServerConfig.Runtime.TransportProtocol) {
 	case mcpfile.TransportProtocolStreamableHttp:
```

Comment:
should we move the cases to separate functions, like `runStreamableHttpServer` ?